mtp_api/apps/core/forms.py

```diff
@@ -128,12 +215,78 @@ def format_date(d):
             return format_date
         return lambda d: d.strftime('%b %Y')
 
+    def group_months_into_periods(self, monthly_rows):
+        if self.cleaned_data['period'] == 'quarterly':
+            yield from self.group_months_into_quarters(monthly_rows)
+        elif self.cleaned_data['period'] == 'yearly':
+            yield from self.group_months_into_years(12, monthly_rows)
+        elif self.cleaned_data['period'] == 'financial':
+            yield from self.group_months_into_years(3, monthly_rows)
+        else:
+            yield from monthly_rows
+
+    def group_months_into_quarters(self, monthly_rows):
```

Comment:
i'm fairly ambivalent about `doctest`s, but not against them. i sort of like the simplicity of standard `unittest`s since they're generally in one place rather than interspersed in the code so are easy to find

Reply:
i doubt they'd play well with and DB functionality so would need to be carefully tailored, right?